libs/state/src/lib/rx-state.service.ts

```diff
@@ -99,6 +99,25 @@ export class RxState<State extends object>
     this.subscription.unsubscribe();
   }
 
+  /**
+   * @description
+   *
+   * Return RxState in ReadOnly mode exposing only get() and select() methods.
+   * This can be helpful when you don't want other's to write in your state.
+   *
+   * @example
+   * ```typescript
+   * const readOnlyState = state.asReadOnly();
+   * const getNum = state.get('num');
+   * const selectNum$ = state.select('num');
+   * ```
+   *
+   * @return Pick<RxState<State>, 'get' | 'select'>
+   */
+  asReadOnly(): Pick<RxState<State>, 'get' | 'select'> {
+    return this as Pick<RxState<State>, 'get' | 'select'>;
```

Comment:
with this approach we only return a new type, the disallowed methods are still available though. It's maybe safer to return a new object, e.g.

```ts
return { get: this.get, select: this.select, computed: this.computed, signal: this.signal }
```